### PR TITLE
Add demo mode to install full stack on VM

### PR DESCRIPTION
**Pull request by cjh1**

### CODE REVIEW

**Comment by cjh1** [comment]
@kotfic PTAL

**Comment by cjh1** [comment]
@TristanWright Can you try this out on your Mac. You should be able to DEMO=1 vagrant up. Then point your browser at http:/localhost:8888, there should be a demo cluster already created that you can run a test job on. The one issue I ran into that you need todo a refresh on cluster page for the cluster to appear? I guess because the client didn't create the cluster it doesn't know about it at first?

**Comment by cjh1** [comment]
@TristanWright Actually, hold off for the moment, we need to cut a girder_client release before you can test this ...

**Comment by cjh1** [comment]
@kotfic @TristanWright This is now ready to testing/review

File: Vagrantfile [hunk 1]
```diff
@@ -81,13 +87,91 @@ Vagrant.configure(2) do |config|
...
+    # Create temp directory
+    dir = Dir.mktmpdir()
+
+    # First download cumulus so we have access to the playbooks
+    config.vm.provision "ansible" do |ansible|
+
+      # If there is a better way to run playbook aganst localhost let
+      # me know!
+      ansible.raw_arguments = ['--limit=localhost',
+        '--inventory-file=demo/inventory/localhost',
```
Comment (kotfic) [inline Vagrantfile]: If you want you can pass in '--inventory-file="localhost,"'  (with the trailing comma) and remove the demo/inventory/localhost file.

**Comment by kotfic** [comment]
Having resolved local issues with running ansible,  I am seeing the following error after provisioning:

```
TASK [fixtures : Wait for cluster to move into the created state] **************
task path: /home/kotfic/tmp/cumulus/HPCCloud-deploy/demo/roles/fixtures/tasks/main.yml:60
fatal: [hpccloud]: FAILED! => {"failed": true, "msg": "The conditional check 'result.json.status == 'created' or result.json.status == 'running'' failed. The error was: error while evaluating conditional (result.json.status == 'created' or result.json.status == 'running'): 'dict object' has no attribute 'status'"}
```

I am running ansible 2.0.1.0

```
(cumulus-deploy)kotfic@minastirith:~/tmp/cumulus/HPCCloud-deploy ⇒ ansible --version
ansible 2.0.1.0 (detached HEAD bb6cadefa2) last updated 2016/05/26 09:23:17 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 262e2a3302) last updated 2016/05/26 09:23:21 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD e0be11da08) last updated 2016/05/26 09:23:22 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

And vagrant 1.8.1

```
(cumulus-deploy)kotfic@minastirith:~/tmp/cumulus/HPCCloud-deploy ⇒ vagrant --version
Vagrant 1.8.1
```

**Comment by cjh1** [comment]
Looks like we need ansible 2.0.2

**Comment by TristanWright** [comment]
I haven't gotten that far yet, starting with a fresh VM I ran into a bunch of auth issues which I fixed in [a branch](https://github.com/Kitware/HPCCloud/pull/415).

File: Vagrantfile (hunk 1 above)
Comment (cjh1) [inline Vagrantfile]: I pushed a fix that updates this.

**Comment by cjh1** [comment]
@TristanWright I was seeing auth issues a well, I will test out your branch

**Comment by cjh1** [comment]
@kotfic Do the last two commit look ok to you?

**Comment by kotfic** [comment]
LGTM

**Comment by TristanWright** [comment]
lgtm ran a simulation on the premade cluster. 
untitled gif

**Pull request merged**